### PR TITLE
Added generic email integration

### PR DESCRIPTION
**Pull request by tarzzz**
A crude implementation of sendmail method using mandrill API. Need to work more on it.

### CODE REVIEW

File: broadgauge/sendmail.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import mandrill
+import default_settings
+
```
Comment (shrayasr) [inline broadgauge/sendmail.py:5]: PEP8. Needs 1 more blank line here

File: broadgauge/default_settings.py [hunk 2]
```diff
@@ -9,4 +9,10 @@
...
 
-db_parameters = dict(dbn='postgres', db='pythonexpress')
\ No newline at end of file
+db_parameters = dict(dbn='postgres', db='pythonexpress')
+
+# These are settings for sending mails
+smtp = "smtp.googlemail.com"
+smtp_credentials = {'login':'from@example.com',
+              'password':'password', 'tls':True}
+from_email = "from_email@example.com"
```
Comment (tarzzz) [inline broadgauge/default_settings.py:18]: @anandology We can put in mandrill smtp, and credentials in default_settings

File: broadgauge/sendmail.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    >>> from sendmail import sendmail
+    >>> sendmail("emails/trainers/welcome.html",to=..."some_email.com", 
+                               sub="Hey Friend!", variable1=var, variable2=var2)
+    Email sent to some_email.com
+    """
+    
+    envelope = Envelope(
+        from_addr=default_settings.from_email,
+        to_addr=kwargs.pop('to'),
+        subject=kwargs.pop('subject'),
```
Comment (tarzzz) [inline broadgauge/sendmail.py:40]: We can give subject as a kwarg, but not sure how to put it in a template, and then utilize it here.

P.S small typo, either should be "subject" or "sub"

**Comment by tarzzz** [comment]
@anandology Please review this one. We can add this to PythonExpress after adding relevant smtp credentials in `default_settings.py`

**Comment by tarzzz** [comment]
@anandology Removed merge conflicts(merged master to this branch)

File: requirements.txt [hunk 4]
```diff
@@ -9,4 +9,5 @@ rauth==0.7.0
 requests==2.3.0
 unittest2==0.5.1
 wsgiref==0.1.2
+envelopes
```
Comment (baijum) [inline requirements.txt]: Please specify version number here:

```
Envelopes==0.4
```

Comment (tarzzz) [inline requirements.txt]: incoming: new commit :-D

**Comment by anandology** [comment]
Thanks @tarzzz. I merged this pull request.

**Pull request closed without merging**